src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -87,6 +88,32 @@ class BasePaymentsPage extends React.Component {
             this.debounceSetShouldShowLoadingSpinner();
         }
 
+        if (this.state.shouldShowDefaultDeleteMenu || this.state.shouldShowPasswordPrompt) {
+            // We should reset selected payment method state values and close corresponding modals if the selected payment method is deleted
+            let shouldResetPaymentMethodData = false;
+
+            if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT && _.isEmpty(this.props.bankAccountList[this.state.methodID])) {
+                // If the selected payment method is bank account and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.DEBIT_CARD && _.isEmpty(this.props.cardList[this.state.methodID])) {
+                // If the selected payment method is card and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            } else if (this.state.selectedPaymentMethodType === CONST.PAYMENT_METHODS.PAYPAL && this.props.payPalMeData !== prevProps.payPalMeData && _.isEmpty(this.props.payPalMeData)) {
+                // If the selected payment method is paypal and it has been deleted, reset the selected payment method data and close corresponding modals
+                shouldResetPaymentMethodData = true;
+            }
+            if (shouldResetPaymentMethodData) {
+                // Close corresponding selected payment method modals which are open
+                if (this.state.shouldShowDefaultDeleteMenu) {
+                    this.hideDefaultDeleteMenu();
+                } else if (this.state.shouldShowPasswordPrompt) {
+                    this.hidePasswordPrompt();
+                } else {
+                    this.resetSelectedPaymentMethodData();
```

Comment:
I think this statement is unreachable no? Testing it here.
Was it meant to be like this?
```diff
if (shouldResetPaymentMethodData) {
    // Close corresponding selected payment method modals which are open
    if (this.state.shouldShowDefaultDeleteMenu) {
        this.hideDefaultDeleteMenu();
    } else if (this.state.shouldShowPasswordPrompt) {
        this.hidePasswordPrompt();
-    } else {
+    }
    this.resetSelectedPaymentMethodData();
-    }
```

Reply:
Initially it was present without if like this for extra protection

                if (this.state.shouldShowDefaultDeleteMenu) {
                    this.hideDefaultDeleteMenu();
                }
                if (this.state.shouldShowPasswordPrompt) {
                    this.hidePasswordPrompt();
                }
                this.resetSelectedPaymentMethodData();

Now it seems unreachable with the new if else. So removed.